xcodeproj/internal/output_files.bzl

```diff
@@ -69,36 +69,58 @@ def _create(
         is_framework = False
         swift = None
 
-    if compiled:
-        # We only need the single swiftmodule in order to download everything
-        # from the remote cache (because of
-        # `--experimental_remote_download_regex`). Reducing the number of items
-        # in an output group keeps the BEP small.
-        closest_compiled = depset(compiled[0:1])
+    if should_produce_output_groups:
+        if compiled:
+            # We only need the single swiftmodule in order to download
+            # everything from the remote cache (because of
+            # `--experimental_remote_download_regex`). Reducing the number of
+            # items in an output group keeps the BEP small.
+            closest_compiled = depset(compiled[0:1])
+        else:
+            closest_compiled = depset(transitive = [
+                info.outputs._closest_compiled
+                for attr, info in transitive_infos
+                if (not info.outputs._is_framework and
+                    (not automatic_target_info or
+                     info.target_type in automatic_target_info.xcode_targets.get(
+                         attr,
+                         [None],
+                     )))
+            ])
+
+        transitive_indexestores = depset(
+            [indexstore] if indexstore else None,
+            transitive = [
+                info.outputs._transitive_indexestores
+                for attr, info in transitive_infos
+                if (not automatic_target_info or
+                    info.target_type in automatic_target_info.xcode_targets.get(
+                        attr,
+                        [None],
+                    ))
+            ],
+        )
+
+        # TODO: Once BwB mode no longer has target dependencies, remove
+        # transitive products. Until then we need them, to allow `Copy Bazel
+        # Outputs` to be able to copy the products of transitive dependencies.
```

Comment:
Does this comment still apply to BwB since it's guarded by `ctx.attr._build_mode == "xcode"` now?

Reply:
`should_produce_output_groups` is `ctx.attr._build_mode == "bazel"` in this file (it's `"xcode"` in the other file).

Reply:
🤦